android/KMEA/app/src/main/java/com/tavultesoft/kmea/ModelPickerActivity.java

```diff
@@ -155,9 +156,19 @@ public void onItemClick(AdapterView<?> parent, View view, final int position, lo
             // While awkward, we must obtain the preInstalledModelMap before any installations occur.
             // We don't want to remove the model we just installed, after all!
             if(preInstalledModelMap != null) {
-              LexicalModel preInstalled = new LexicalModel(preInstalledModelMap);
+              // This might be unncessary
+              LexicalModel preInstalled = new LexicalModel(
+                preInstalledModelMap.get(KMManager.KMKey_PackageID),
+                preInstalledModelMap.get(KMManager.KMKey_LexicalModelID),
+                preInstalledModelMap.get(KMManager.KMKey_LexicalModelName),
+                preInstalledModelMap.get(KMManager.KMKey_LanguageID),
+                preInstalledModelMap.get(KMManager.KMKey_LanguageName),
+                preInstalledModelMap.get(KMManager.KMKey_LexicalModelVersion),
+                preInstalledModelMap.get(KMManager.KMKey_HelpLink),
+                "" // String modelURL
+                );
               String itemKey = String.format("%s_%s_%s",
-                  preInstalled.map.get(KMManager.KMKey_PackageID), preInstalled.getLanguageCode(), preInstalled.getResourceId());
+                  preInstalled.getPackageID(), preInstalled.getLanguageCode(), preInstalled.getResourceID());
```

Comment:
I'm also pondering adding a `Keyboard.getItemKey()` method that returns this formatted string...